arangod/Aql/RemoteExecutor.h

```diff
@@ -121,22 +102,15 @@ class ExecutionBlockImpl<RemoteExecutor> : public ExecutionBlock {
 
   /// @brief the last unprocessed result. Make sure to reset it
   ///        after it is processed.
-  std::shared_ptr<httpclient::SimpleHttpResult> _lastResponse;
+  std::unique_ptr<fuerte::Response> _lastResponse;
 
   /// @brief the last remote response Result object, may contain an error.
   arangodb::Result _lastError;
 
-  /// @brief Mutex to cover against the race, that a getSome request
-  ///        is responded before the ticket id is registered.
-  arangodb::Mutex _communicationMutex;
-#ifdef ARANGODB_USE_GOOGLE_TESTS
-  std::atomic<std::thread::id> _communicationMutexOwner;  // current thread owning '_communicationMutex' lock (workaround for non-recusrive MutexLocker)
-#endif
-
-  OperationID _lastTicketId;
-
   bool _hasTriggeredShutdown;
 
+  std::atomic<unsigned> _lastTicket;  /// used to check for canceled requests
```

Comment:
Suggestion: move the `_lastTicket` member before the two bools, so we can save a bit of space.
Though it is not really important here.